Doc/c-api/module.rst

```diff
@@ -196,23 +196,41 @@ or request "multi-phase initialization" by returning the definition struct itsel
    .. c:member:: traverseproc m_traverse
 
       A traversal function to call during GC traversal of the module object, or
-      ``NULL`` if not needed. This function may be called before module state
-      is allocated (:c:func:`PyModule_GetState()` may return `NULL`),
-      and before the :c:member:`Py_mod_exec` function is executed.
+      ``NULL`` if not needed.
+
+      This function is not called before the module state is allocated (before
+      the :c:member:`Py_mod_exec` function is executed): it is not called if
+      :c:member:`m_size` is greater than 0 and module state
+      (:c:func:`PyModule_GetState`) is ``NULL``.
+
+      .. versionchanged:: 3.9
+         It is no longer called before the module state is allocated.
```

Comment:
IMO this needs clarification.
If a module has `m_size` of 0 or -1, so the state is not allocated, then `m_traverse` *can* be called before the `Py_mod_exec` function. Right?

```suggestion
      This function is not called if module state was requested but is not allocated.
      (This is the case immediately after the module is created, before the
      :c:member:`Py_mod_exec` function is executed.)
      More precisely, this function is not called if :c:member:`m_size` is positive and
      module state (as returned by :c:func:`PyModule_GetState`) is ``NULL``.

      .. versionchanged:: 3.9
         No longer called before the module state is allocated.```